src/Chain/Toolbox/Tool/Crawler.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+declare(strict_types=1);
+
+namespace PhpLlm\LlmChain\Chain\Toolbox\Tool;
+
+use PhpLlm\LlmChain\Chain\Toolbox\Attribute\AsTool;
+use Symfony\Component\DomCrawler\Crawler as DomCrawler;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+#[AsTool('crawler', 'A tool that crawls one page of a website and returns the visible text of it.')]
+final readonly class Crawler
+{
+    public function __construct(
+        private HttpClientInterface $httpClient,
+    ) {
+        if (!class_exists(DomCrawler::class)) {
+            throw new \RuntimeException('The DomCrawler component is not installed. Please install it using "composer require symfony/dom-crawler".');
```

Comment:
If we have an own runtime exception we should use it. Cannot check, I am currently on a phone 📱

Reply:
oh the exception usage is quite inconsistent - will tackle in a follow-up and streamline all places :+1: